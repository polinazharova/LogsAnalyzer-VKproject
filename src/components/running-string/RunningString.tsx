import RunningStringPattern from "../../assets/running_string_pattern.svg";
import "./RunningString.styles.scss";

interface RunningStringProps {
  children: React.ReactNode;
}

const RunningString: React.FC<RunningStringProps> = ({ children }) => {
  return (
    <div className="main__running-string">
      <div className="main__running-string-content">
        <img
          className="main__running-string-pattern"
          src={RunningStringPattern}
          alt="Running string background image"
        />
        <span className="main__running-string-text">{children}</span>
        <img
          className="main__running-string-pattern"
          src={RunningStringPattern}
          alt="Running string background image"
        />
      </div>
      <div className="main__running-string-content">
        <img
          className="main__running-string-pattern"
          src={RunningStringPattern}
          alt="Running string background image"
        />
        <span className="main__running-string-text">{children}</span>
        <img
          className="main__running-string-pattern"
          src={RunningStringPattern}
          alt="Running string background image"
        />
      </div>
    </div>
  );
};

export default RunningString;
