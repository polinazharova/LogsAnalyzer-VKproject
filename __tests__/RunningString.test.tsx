import React from "react";
import RunningString from "../src/components/running-string/RunningString";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("Running string component", () => {
  test("renders children correctly", () => {
    render(<RunningString>RS</RunningString>);
    const rs = screen.getAllByText("RS");
    rs.forEach((elem) => {
      expect(elem).toBeInTheDocument();
    });
  });

  test("matches snapshot", () => {
    const { container } = render(<RunningString>RS</RunningString>);
    expect(container).not.toBeEmptyDOMElement();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("applies correct className", () => {
    render(<RunningString>RS</RunningString>);
    const rs = screen.getAllByText("RS");
    rs.forEach((elem) => {
      expect(elem).toHaveClass("main__running-string-text");
    });
  });
});
