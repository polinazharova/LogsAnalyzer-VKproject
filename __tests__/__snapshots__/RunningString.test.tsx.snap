// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running string component matches snapshot 1`] = `
<div
  class="main__running-string"
>
  <div
    class="main__running-string-content"
  >
    <img
      alt="Running string background image"
      class="main__running-string-pattern"
      src=""
    />
    <span
      class="main__running-string-text"
    >
      RS
    </span>
    <img
      alt="Running string background image"
      class="main__running-string-pattern"
      src=""
    />
  </div>
  <div
    class="main__running-string-content"
  >
    <img
      alt="Running string background image"
      class="main__running-string-pattern"
      src=""
    />
    <span
      class="main__running-string-text"
    >
      RS
    </span>
    <img
      alt="Running string background image"
      class="main__running-string-pattern"
      src=""
    />
  </div>
</div>
`;
